de-lib/src/main/java/org/iplantc/de/teams/client/presenter/EditTeamPresenterImpl.java

```diff
@@ -0,0 +1,302 @@
+                progressDlg.hide();
+            }
+
+            @Override
+            public void onSuccess(List<UpdateMemberResult> updateMemberResults) {
+                hideable.hide();
+                progressDlg.updateProgress();
+                view.unmask();
+                ensureHandlers().fireEvent(new TeamSaved(group));
+            }
+        });
+    }
+
+    void addPrivilegesToTeam(Group group, List<Privilege> privileges, IsHideable hideable) {
+        progressDlg.updateProgress();
+
+        List<UpdatePrivilegeRequest> updateList = convertPrivilegesToUpdateRequest(privileges);
+
+        UpdatePrivilegeRequestList allUpdates = factory.getUpdatePrivilegeRequestList().as();
+        allUpdates.setRequests(updateList);
+
+        serviceFacade.updateTeamPrivileges(group, allUpdates, new AsyncCallback<List<Privilege>>() {
+            @Override
+            public void onFailure(Throwable throwable) {
+                ErrorHandler.post(throwable);
+                mode = EditTeamView.MODE.EDIT;
+                view.unmask();
+                progressDlg.hide();
+            }
+
+            @Override
+            public void onSuccess(List<Privilege> privileges) {
+                addMembersToTeam(group, hideable);
+            }
+        });
+    }
+
+    Subject createSelfSubject() {
+        Subject subject = factory.getSubject().as();
+        subject.setId(userInfo.getUsername());
+        return subject;
+    }
+
+    @Override
+    public void onUserSearchResultSelected(UserSearchResultSelected userSearchResultSelected) {
+        Subject subject = userSearchResultSelected.getSubject();
+        String tag = userSearchResultSelected.getTag();
+        Privilege privilege = factory.getPrivilege().as();
+        privilege.setSubject(subject);
+        privilege.setPrivilegeType(PrivilegeType.read);
+
+        if (SEARCH_MEMBERS_TAG.equals(tag)) {
+            view.addMembers(Lists.newArrayList(privilege));
+        } else {
+            view.addNonMembers(Lists.newArrayList(privilege));
+        }
+
+    }
+
+    @Override
+    public void onRemoveMemberPrivilegeSelected(RemoveMemberPrivilegeSelected event) {
+        Privilege privilege = event.getPrivilege();
+        if (EditTeamView.MODE.CREATE == mode) {
+            view.removeMemberPrivilege(privilege);
+        }
+    }
+
+    @Override
+    public void onRemoveNonMemberPrivilegeSelected(RemoveNonMemberPrivilegeSelected event) {
+        Privilege privilege = event.getPrivilege();
+        if (EditTeamView.MODE.CREATE == mode) {
+            view.removeNonMemberPrivilege(privilege);
+        }
+    }
+
+    @Override
+    public void onAddPublicUserSelected(AddPublicUserSelected event) {
+        addPublicUser();
+    }
+
+    List<UpdatePrivilegeRequest> convertPrivilegesToUpdateRequest(List<Privilege> privileges) {
+        List<UpdatePrivilegeRequest> updateList = Lists.newArrayList();
+
+        privileges.forEach(new Consumer<Privilege>() {
```

Comment:
Actually no, I couldn't figure out how to do this with a stream 😄   I'll look into it again though because I do prefer that syntax over `foreach`.